sklearn/ensemble/tests/test_forest.py

```diff
@@ -1786,3 +1788,25 @@ def test_base_estimator_property_deprecated(name):
     )
     with pytest.warns(FutureWarning, match=warn_msg):
         model.base_estimator_
+
+
+def test_read_only_buffer(monkeypatch):
+    """RandomForestClassifier must work on readonly sparse data.
+
+    Non-regression test for: https://github.com/scikit-learn/scikit-learn/issues/25333
+    """
+    monkeypatch.setattr(
+        sklearn.ensemble._forest,
+        "Parallel",
+        partial(Parallel, max_nbytes=100),
+    )
+    X, y = make_classification(100, n_features=200)
+    X = csr_matrix(X, copy=True)
+
+    # Create memmapped readonly buffer
+    (X.data, X.indices, X.indptr, y) = create_memmap_backed_data(
+        (X.data, X.indices, X.indptr, y)
+    )
+
+    clf = RandomForestClassifier(n_jobs=-1)
+    cross_val_score(clf, X, y)
```

Comment:
Minor changes to reduce the run time of the tests, set the `random_seed` to be deterministic, and I do not think the memmap here is required for the test to fail on `main`:

```suggestion
    X, y = make_classification(n_samples=100, n_features=20, random_state=0)
    X = csr_matrix(X, copy=True)

    clf = RandomForestClassifier(n_jobs=2, random_state=0)
    cross_val_score(clf, X, y, cv=2)
```